tasks/20230526-gauge-working-balance-helper/test/L1TimeDecay.fork.ts

```diff
@@ -0,0 +1,192 @@
+
+    // Create the Gauge, vs using an existing one, so that we can control the total balance
+    const tx = await factory.create(LP_TOKEN, weightCap);
+    const event = expectEvent.inReceipt(await tx.wait(), 'GaugeCreated');
+
+    gauge = await gaugeFactoryTask.instanceAt('LiquidityGaugeV5', event.args.gauge);
+    expect(await gauge.lp_token()).to.equal(LP_TOKEN);
+
+    expect(await factory.isGaugeFromFactory(gauge.address)).to.be.true;
+  });
+
+  const stakeAmount = fp(100);
+
+  before('stake in gauge', async () => {
+    await lpToken.connect(lpTokenHolder).transfer(veBALHolder.address, stakeAmount);
+    await lpToken.connect(lpTokenHolder).transfer(other.address, stakeAmount);
+
+    await lpToken.connect(lpTokenHolder).approve(gauge.address, MAX_UINT256);
+    await lpToken.connect(veBALHolder).approve(gauge.address, MAX_UINT256);
+    await lpToken.connect(other).approve(gauge.address, MAX_UINT256);
+
+    await gauge.connect(lpTokenHolder)['deposit(uint256)'](stakeAmount.mul(100));
+    await gauge.connect(veBALHolder)['deposit(uint256)'](stakeAmount);
+    await gauge.connect(other)['deposit(uint256)'](stakeAmount);
+  });
+
+  it('projected balance should equal current', async () => {
+    const [currentWorkingBalance, projectedWorkingBalance] = await workingBalanceHelper.getWorkingBalances(
+      gauge.address,
+      veBALHolder.address
+    );
+
+    // Ensure we have equal balances (that are non-zero)
+    expect(projectedWorkingBalance).to.eq(currentWorkingBalance);
+    expect(projectedWorkingBalance).to.gt(0);
+  });
+
+  context('with veBAL', () => {
+    let bal80weth20Pool: Contract;
+    let whaleBalance: BigNumber;
+
+    before('create veBAL whale', async () => {
+      const [poolAddress] = await vault.getPool(VEBAL_POOL_ID);
+
+      bal80weth20Pool = await instanceAt('IERC20', poolAddress);
+
+      await vault.connect(veBALHolder).joinPool(
+        VEBAL_POOL_ID,
+        veBALHolder.address,
+        veBALHolder.address,
+        {
+          assets: [BAL, ZERO_ADDRESS],
+          maxAmountsIn: [0, VAULT_BOUNTY],
+          fromInternalBalance: false,
+          userData: WeightedPoolEncoder.joinExactTokensInForBPTOut([0, VAULT_BOUNTY], 0),
+        },
+        { value: VAULT_BOUNTY }
+      );
+
+      const totalBalance = await bal80weth20Pool.balanceOf(veBALHolder.address);
+      whaleBalance = fpMul(totalBalance, fp(0.99));
+      const otherBalance = totalBalance.sub(whaleBalance);
+
+      await bal80weth20Pool.connect(veBALHolder).transfer(other.address, otherBalance);
+
+      await bal80weth20Pool.connect(veBALHolder).approve(votingEscrow.address, MAX_UINT256);
+      await bal80weth20Pool.connect(other).approve(votingEscrow.address, MAX_UINT256);
+
+      const currentTime = await currentTimestamp();
+      await votingEscrow.connect(other).create_lock(otherBalance, currentTime.add(LOCK_PERIOD));
+    });
+
+    it('veBAL decay affects projected balances', async () => {
```

Comment:
Is this testing the veBAL decay, or the fact that a whale created the lock between the "before" and the "after" stages?

The share for `other` should go down even without advancing a week, since the veBAL whale increases the total supply and hence reduces the share of `other` right?

Reply:
Yes; we probably don't even need this test, but I renamed it to better describe this. (Don't need the time decay.)